components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/dao/impl/DeviceOrganizationDAOImpl.java

```diff
@@ -0,0 +1,456 @@
+            stmt.setString(1, deviceName);
+            stmt.setString(2, deviceId);
+            rows = stmt.executeUpdate();
+            if (rows > 0) {
+                updatedName = deviceName;
+            }
+        } catch (SQLException e) {
+            throw new DeviceOrganizationDAOException("Error occurred while updating Name of device '" +
+                    deviceId + "'", e);
+        } finally {
+            DeviceManagementDAOUtil.cleanupResources(stmt, null);
+            return updatedName;
+        }
+    }
+
+    /**
+     * Update device organization path
+     *
+     * @param deviceId unique device identifier
+     * @param parent   parent that device is child to in the network
+     * @return the updated device path
+     * @throws DeviceManagementDAOException
+     */
+    @Override
+    public String updateDeviceOrganizationParent(String deviceId, String parent) throws DeviceOrganizationDAOException {
+        Connection conn;
+        PreparedStatement stmt = null;
+        int rows;
+        String updatedParent = null;
+        try {
+            conn = this.getConnection();
+            String sql = "UPDATE DEVICE_ORGANIZATION_MAP SET DEVICE_PARENT = ? WHERE DEVICE_ID = ?";
+            stmt = conn.prepareStatement(sql, new String[]{"id "});
+            stmt.setString(1, parent);
+            stmt.setString(2, deviceId);
+            rows = stmt.executeUpdate();
+            if (rows > 0) {
+                updatedParent = parent;
+            }
+        } catch (SQLException e) {
+            throw new DeviceOrganizationDAOException("Error occurred while path of updating device '" +
+                    deviceId + "'", e);
+        } finally {
+            DeviceManagementDAOUtil.cleanupResources(stmt, null);
+            return updatedParent;
+        }
+    }
+
+    /**
+     * This method allows to update the no. of minutes since last contact with device
+     *
+     * @param deviceId    unique device identifier
+     * @param newPingMins number of minutes since last ping from device
+     * @return
+     * @throws DeviceOrganizationDAOException
+     */
+    @Override
+    public int updateDevicePingMins(String deviceId, int newPingMins) throws DeviceOrganizationDAOException {
+        Connection conn;
+        PreparedStatement stmt = null;
+        int rows;
+        int updatedPingMins = -1;
+        try {
+            conn = this.getConnection();
+            String sql = "UPDATE DEVICE_ORGANIZATION_MAP SET MINUTES_SINCE_LAST_PING = ? WHERE DEVICE_ID = ?";
+            stmt = conn.prepareStatement(sql, new String[]{"id"});
+            stmt.setInt(1, newPingMins);
+            stmt.setString(2, deviceId);
+            rows = stmt.executeUpdate();
+            if (rows > 0) {
+                updatedPingMins = newPingMins;
+            }
+        } catch (SQLException e) {
+            throw new DeviceOrganizationDAOException("Error occurred while updating No. of mins since last ping of device '" +
```

Comment:
Please check whether this line exceeds the column length. And check this on all the other code lines as well.

Reply:
Fixed e1506f6a0eee9a3d32b4d22df9409b50221decc8